ouroboros-consensus-cardano-tools/block-tool.md

```diff
@@ -0,0 +1,102 @@
+# Block Inspector
+
+`block-tool` is a command-line utility that can dump the content of a full block to `stdout` in JSON. The block can either:
+
+* Be given as standalone hex-CBOR encoded data, either to its standard input or from a file,
+* Be read from an existing _immutable_ DB.
+
+## Rationale
+
+There already exists tools providing human-readable representation of the content of a block:
+
+* Publicly available explorers like [Cardano Explorer](https://explorer.cardano.org/), [Cardanoscan](https://cardanoscan.io/), ...
+* Services like [Blockfrost](https://blockfrost.io/), [Ogmios](https://ogmios.dev/), [scrolls](https://github.com/txpipe/scrolls), ...
+
+This tool fits some more specialised use cases:
+
+* Inspecting data directly from an existing node's DB which might live on a different network,
+* Dumping content of some random block that might  not even exist on the chain anymore,
+* Debugging output of consensus code.
+
+## Usage
+
+```
+Usage: block-tool [--file-in FILE | --db-directory DIR --config FILE
```

Comment:
Would it make sense to refer to the output of `block-tool -h` to avoid repeating the text here?

Reply:
Sort of, yes. My goal here was more to convey a sense of what this tool looks like.